mesonbuild/coredata.py

```diff
@@ -248,7 +353,7 @@ def __init__(self, options):
         self.cross_files = self.__load_config_files(options.cross_file, 'cross')
         self.compilers = OrderedDict()
         self.cross_compilers = OrderedDict()
-        self.deps = OrderedDict()
+        self.deps = DependencyCache(self)
```

Comment:
I think this should be a `PerMachine`, especially if #5263 (and maybe also #4010) can soon land?

I kind of rather land those first and then do this as refactoring caching stuff can be tricky, but writing it after build and host are cleanly separated should be a lot easier.

Reply:
OK now that I finished the read, I take a bit back, the rebase won't be too bad. But still it would be good to get those in so we all have easier times writing things like this in the first place.